app/views/posts/index.html.erb

```diff
@@ -1 +1,40 @@
-<h1>User 1 all posts</h1>
+<div class="container">
+  <!-- User Information Section -->
+  <div class="user_info">
+    <div class="img">
+      <!-- Display user photo using Rails image_tag helper -->
+      <%= image_tag @user.photo, alt: "User Photo", class: "user-photo" %>
+    </div>
+    <div class="user">
+      <!-- Display user name with a link to user profile -->
+      <h2><%= link_to @user.name, user_path(@user) %></h2>
```

Comment:
- [ ] Please make sure your `posts index` page shows without errors, for now, this error  `undefined method 'photo' for nil:NilClass` is shown when I try to `post index` page it means the `@user` is not defined in your `post controller` file so kindly make sure you create `@user` variable and this is error will be gone :+1:  check below the screenshot of error :slightly_smiling_face: 

![Screenshot from 2023-12-15 22-04-58](https://github.com/MohammadYaser/blog-app/assets/45226848/8be7791d-6111-446b-bdda-764909b89049)

**Suggestions :point_down:** 
```rb
class PostsController < ApplicationController
  def index
    @posts = Post.where(author_id: params[:user_id])
    @user = User.find(params[:user_id])
  end

  def show
    @post = Post.find(params[:id])
    @user = User.find(@post.author_id)
  end
end

```